tests/unit/task/test_print.py

```diff
@@ -0,0 +1,156 @@
+"""Test that a task tree can be printed in a human readable form."""
+import copy
+import textwrap
+
+import pytest
+
+from esmvalcore._task import DiagnosticTask
+from esmvalcore.preprocessor import PreprocessingTask, PreprocessorFile
+
+
+@pytest.fixture
+def preproc_file():
+    return PreprocessorFile(
+        attributes={'filename': '/output/preproc/file.nc'},
+        settings={
+            'extract_levels': {
+                'scheme': 'linear',
+                'levels': [95000]
+            },
+        },
+    )
+
+
+@pytest.fixture
+def preproc_task(preproc_file):
+    return PreprocessingTask(products=[preproc_file])
+
+
+@pytest.fixture
+def diagnostic_task(monkeypatch):
+    script = '/some/where/esmvaltool/diag_scripts/test.py'
+    settings = {
+        'run_dir': str('/output/run'),
+    }
+
+    monkeypatch.setattr(DiagnosticTask, '_initialize_cmd', lambda self: None)
```

Comment:
Do you think it would be possible to make the test without using monkeypatch?

We use monkeypatch a lot in the tests, but I find such code is often difficult to debug and it makes it follow the code. This introduces a very tight coupling between the implementation and the patch. This makes the test fragile. Do you think it would be possible to initialize `DiagnosticTask` directly (maybe via some factory method or constructor)?

Reply:
Done.

I agree that there are downsides to mocking, but also good things: mocking can make it easier to find bugs if a test fails, because it helps to test smaller chunks at a time and it is also useful to speed up tests that require I/O (as was the case here with creating a file).